src/main/scala/com/databricks/labs/overwatch/pipeline/Migration.scala

```diff
@@ -0,0 +1,184 @@
+package com.databricks.labs.overwatch.pipeline
+
+import com.databricks.dbutils_v1.DBUtilsHolder.dbutils
+import com.databricks.labs.overwatch.env.{Database, Workspace}
+import com.databricks.labs.overwatch.utils._
+import org.apache.log4j.{Level, Logger}
+import com.databricks.labs.overwatch.utils.Helpers.removeTrailingSlashes
+
+import java.io.FileNotFoundException
+
+
+class Migration(_sourceETLDB: String, _targetPrefix: String, _workspace: Workspace, _database: Database, _config: Config)
+  extends Pipeline(_workspace, _database, _config){
+
+
+  import spark.implicits._
+  private val migrateRootPath = removeTrailingSlashes(_targetPrefix)
+  private val workSpace = _workspace
+  private val sourceETLDB = _sourceETLDB
+  private val bronze = Bronze(workSpace)
+  private val silver = Silver(workSpace)
+  private val gold = Gold(workSpace)
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  private val driverCores = java.lang.Runtime.getRuntime.availableProcessors()
+  val Config = _config
+
+  private[overwatch] def buildCloneSpecs(
```

Comment:
This function is created repeatedly as private funds. Please generalize this func and move it to where it makes most sense. It should likely be private[overwatch] but it should only be defined in one place.